nbi/engine/native/launcher/windows/src/JavaUtils.c

```diff
@@ -125,58 +125,79 @@ DWORD isJavaCompatible(JavaProperties *currentJava, JavaCompatible ** compatible
 JavaVersion * getJavaVersionFromString(char * string, DWORD * result) {
     JavaVersion *vers = NULL;
     if(getLengthA(string)>=3) {
-        //hope that at least we "major.minor" : 1.5
-        if(string[1]=='.') {
-            char c = string[0];
+        char *p = string;
+        
+        // get major 
+        long major = 0;
+        while(p!=NULL) {
+            char c = p[0];
```

Comment:
`*p++` (allows to remove `p++` from the below branches) or `*p` (the same as `p[0]` but does not provoke the reader to think about array access)